gemrb/core/EffectQueue.cpp

```diff
@@ -469,22 +460,19 @@ bool EffectQueue::RemoveEffect(const Effect* fx)
 void EffectQueue::ApplyAllEffects(Actor* target) const
 {
 	for (auto fx : effects) {
```

Comment:
W.r.t. the sound effects hanging around to much, I suspect this to be one place where we have a copy of the effect. I found that the effect that is being assigned the `FX_DURATION_JUST_EXPIRED` is not the same as the one from the list.